File: tags/v_3_0_1/aon-registry/src/main/java/com/code/aon/registry/RecordData.java
package com.code.aon.registry;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.code.aon.common.ITransferObject;

@Entity
@Table(name="record_data")
public class RecordData implements ITransferObject {
	
	private Integer id;
	
	private Registry registry;
	
	private Date creationDate;
	
	private String description;
	
	private String notary;
	
	private String number;
	
	private Date recordDate;
	
	private String volume;
	
	private String section;
	
	private String page;
	
	private String sheet;
	
	private String registration;
	
	private RegistryAttachment attach; 

	@Id
	@GeneratedValue
	@Column(nullable=false)
	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	@ManyToOne
	@JoinColumn(name="registry", nullable=false)
	public Registry getRegistry() {
		return registry;
	}

	public void setRegistry(Registry registry) {
		this.registry = registry;
	}

	@Column(name="creation_date")
	public Date getCreationDate() {
		return creationDate;
	}

	public void setCreationDate(Date creationDate) {
		this.creationDate = creationDate;
	}

	@Column(length=64)
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	@Column(length=64)
	public String getNotary() {
		return notary;
	}

	public void setNotary(String notary) {
		this.notary = notary;
	}

	@Column(length=16)
	public String getNumber() {
		return number;
	}

	public void setNumber(String number) {
		this.number = number;
	}

	@Column(name="record_date")
	public Date getRecordDate() {
		return recordDate;
	}

	public void setRecordDate(Date recordDate) {
		this.recordDate = recordDate;
	}

	@Column(length=16)
	public String getVolume() {
		return volume;
	}

	public void setVolume(String volume) {
		this.volume = volume;
	}

	@Column(length=16)
	public String getSection() {
		return section;
	}

	public void setSection(String section) {
		this.section = section;
	}

	@Column(length=16)
	public String getPage() {
		return page;
	}

	public void setPage(String page) {
		this.page = page;
	}

	@Column(length=16)
	public String getSheet() {
		return sheet;
	}

	public void setSheet(String sheet) {
		this.sheet = sheet;
	}

	@Column(length=64)
	public String getRegistration() {
		return registration;
	}

	public void setRegistration(String registration) {
		this.registration = registration;
	}

	@ManyToOne
	@JoinColumn(name="attach")
	public RegistryAttachment getAttach() {
		return attach;
	}

	public void setAttach(RegistryAttachment attach) {
		this.attach = attach;
	}
}

File: aon-web-aio/src/main/webapp/models/registry/RecordData.js
import * as LS from '../../services/localStorageService.js';
import { RecordDataType, CommercialRegistryCode } from './RecordDataEnums.js';

export class RecordData {

    id;
    domain;
    registry;
    creationDate;
    description;
    notary;
    number;
    recordDate;
    volume;
    section;
    page;
    sheet;
    registration;
    attach;
    type;
    irus;
    commercialRegistryCode;

    constructor(recordData) {
        const d = recordData || {};
        this.id = d.id;
        this.domain = d.domain || LS.getDomainId();
        this.registry = d.registry;
        this.creationDate = d.creationDate;
        this.description = d.description || 'Escritura de constitución';
        this.notary = d.notary || '';
        this.number = d.number || '';
        this.recordDate = d.recordDate;
        this.volume = d.volume || '';
        this.section = d.section || '';
        this.page = d.page || '';
        this.sheet = d.sheet || '';
        this.registration = d.registration || 'Inscrita en el Registro mercantil';
        this.attach = d.attach;
        this.type = d.type ? RecordDataType.safeValueOfName(d.type) : RecordDataType.OTHER_REGISTRATIONS;
        this.irus = d.irus || '';
        this.commercialRegistryCode = CommercialRegistryCode.safeValueOfCode(d.commercialRegistryCode);
    }

    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
        return this;
    }

    getDomain() {
        return this.domain;
    }

    setDomain(domain) {
        this.domain = domain;
        return this;
    }

    getRegistry() {
        return this.registry;
    }

    setRegistry(registry) {
        this.registry = registry;
        return this;
    }

    getCreationDate() {
        return this.creationDate;
    }

    setCreationDate(creationDate) {
        this.creationDate = creationDate;
        return this;
    }

    getRecordDate() {
        return this.recordDate;
    }

    setRecordDate(recordDate) {
        this.recordDate = recordDate;
        return this;
    }

    getDescription() {
        return this.description;
    }

    setDescription(description) {
        this.description = description;
        return this;
    }

    getNotary() {
        return this.notary;
    }

    setNotary(notary) {
        this.notary = notary;
        return this;
    }

    getNumber() {
        return this.number;
    }

    setNumber(number) {
        this.number = number;
        return this;
    }

    getVolume() {
        return this.volume;
    }

    setVolume(volume) {
        this.volume = volume;
        return this;
    }

    getSection() {
        return this.section;
    }

    setSection(section) {
        this.section = section;
        return this;
    }

    getPage() {
        return this.page;
    }

    setPage(page) {
        this.page = page;
        return this;
    }

    getSheet() {
        return this.sheet;
    }

    setSheet(sheet) {
        this.sheet = sheet;
        return this;
    }

    getRegistration() {
        return this.registration;
    }

    setRegistration(registration) {
        this.registration = registration;
        return this;
    }

    getAttach() {
        return this.attach;
    }

    setAttach(attach) {
        this.attach = attach;
        return this;
    }

    getType() {
        return this.type;
    }

    setType(type) {
        this.type = type;
        return this;
    }

    getIrus() {
        return this.irus;
    }

    setIrus(irus) {
        this.irus = irus;
        return this;
    }

    getCommercialRegistryCode() {
        return this.commercialRegistryCode;
    }

    setCommercialRegistryCode(commercialRegistryCode) {
        this.commercialRegistryCode = commercialRegistryCode;
        return this;
    }

    toJSON() {
        return {
            id: this.id,
            domain: this.domain,
            registry: this.registry,
            creationDate: this.creationDate,
            description: this.description,
            notary: this.notary,
            number: this.number,
            recordDate: this.recordDate,
            volume: this.volume,
            section: this.section,
            page: this.page,
            sheet: this.sheet,
            registration: this.registration,
            attach: this.attach,
            type: this.type ? this.type.name : null,
            irus: this.irus,
            commercialRegistryCode: this.commercialRegistryCode ? this.commercialRegistryCode.code : null
        }
    }

}
